components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -40,35 +40,55 @@ const updateBadgeTextAllWindows = (windows: chrome.windows.Window[], state?: Rew
 export const rewardsPanelReducer: Reducer<RewardsExtension.State | undefined> = (state: RewardsExtension.State, action: any) => {
   const payload = action.payload
   switch (action.type) {
+    case types.TOGGLE_ENABLE_MAIN: {
+      if (state.initializing && state.enabledMain) {
+        break
+      }
+
+      state = { ...state }
+      const key = 'enabledMain'
+      const enable = action.payload.enable
+      state.initializing = true
+
+      state[key] = enable
+      chrome.braveRewards.saveSetting(key, enable ? '1' : '0')
```

Comment:
Elsewhere in our code, I see us calling `chrome.send('brave_rewards.saveSetting', [key, enable.toString()])`. Might be good to be consistent, unless those are two different use cases?

Reply:
yeah tried to do toString, but was failing, will try again